src/platform/plugins/shared/console/public/application/containers/editor/monaco_editor.tsx

```diff
@@ -123,6 +158,89 @@ export const MonacoEditor = ({
     return actionsProvider.current!.isKbnRequestSelected();
   }, []);
 
+  const copyText = async (text: string) => {
+    if (window.navigator?.clipboard) {
+      await window.navigator.clipboard.writeText(text);
+      return;
+    }
+    throw new Error('Could not copy to clipboard!');
+  };
+
+  // This function will convert all the selected requests to the language by
+  // calling convertRequestToLanguage and then copy the data to clipboard.
+  const copyAs = async (language?: string) => {
+    // Get the language we want to convert the requests to
+    const withLanguage = language || currentLanguage;
+    // Get all the selected requests
+    const requests = await getRequestsCallback();
+
+    // If we have any kbn requests, we should not allow the user to copy as
+    // anything other than curl
+    const hasKbnRequests = requests.some((req) => req.url.startsWith(KIBANA_API_PREFIX));
+
+    if (hasKbnRequests && withLanguage !== 'curl') {
+      toasts.addDanger({
+        title: i18n.translate('console.consoleMenu.copyAsMixedRequestsMessage', {
+          defaultMessage: 'Kibana requests can only be copied as curl',
```

Comment:
@petegaleotti @saikatsarkar056 we seem to have shifted towards saying "copy to" instead of "copy as", I don't have a strong opinion on that change, but might want to consider aligning this message to the new approach :)